### PR TITLE
Migrate to APIv4

### PR DESCRIPTION
**Pull request by tobiasrohloff**
Deprecated stuff:
- `title` in `videos` (use `title` from `course-items`)
- `slug` in `channels` (use `id`)
- `max_points` in `quizzes` (use `max_points` from `course-items`)
- `first_name` and `last_name` from `user-profile` (use `full_name`)
- `section-progresses` relation in `course-progresses` (use `section_progresses`)
- `course-progress` relation in `section-progresses` (use `course_progress`)

### CODE REVIEW

File: app/src/main/java/de/xikolo/models/Quiz.java [hunk 1]
```diff
@@ -47,7 +45,6 @@ public Quiz convertToRealmObject() {
             quiz.instructions = instructions;
             quiz.timeLimit = timeLimit;
             quiz.allowedAttempts = allowedAttempts;
-            quiz.maxPoints = maxPoints;
```
Comment (bengelhaupt) [inline app/src/main/java/de/xikolo/models/Quiz.java:50]: You forgot to remove lines 35 and 36 in the `Quiz.JsonModel`

```
@Json(name = "max_points")
public int maxPoints;
```

Comment (tobiasrohloff) [reply app/src/main/java/de/xikolo/models/Quiz.java:50]: Good catch! Fixed in https://github.com/openHPI/xikolo-android/commit/2b3b5e6420a5e1dbbce44293bb9e5038201a2d89.

**Comment by bengelhaupt** [comment]
This code in `ChannelDao` is causing problems because it still accesses `slug`. The App crashes when opening channel details.
```
override fun find(id: String?): LiveData<Channel> =
        query()
            .beginGroup()
                .equalTo("id", id)
                .or()
                .equalTo("slug", id)
            .endGroup()
            .findFirstAsync()
            .asLiveData()
```

File: app/src/main/java/de/xikolo/controllers/main/MainActivity.kt [hunk 2]
```diff
@@ -284,14 +280,9 @@ class MainActivity : ViewModelActivity<NavigationViewModel>(), NavigationView.On
...
-                headerView.findViewById<TextView>(R.id.textName).text = String.format(
-                    getString(R.string.user_name),
-                    profile?.firstName ?: "",
-                    profile?.lastName ?: ""
-                )
-
-                headerView.findViewById<TextView>(R.id.textEmail).text = profile?.email ?: ""
+                headerView.findViewById<TextView>(R.id.textName).text = user.name
+
+                headerView.findViewById<TextView>(R.id.textEmail).text = user.profile.email
```
Comment (bengelhaupt) [inline app/src/main/java/de/xikolo/controllers/main/MainActivity.kt:285]: `user.profile` can be null in some cases. I added the null check once as a fix. Is there a particular reason you removed it?

**Pull request merged**